lib/engine/game/g_steam_over_holland/entities.rb

```diff
@@ -116,6 +116,8 @@ module Entities
                 type: 'train_discount',
                 discount: 0.1,
                 trains: %w[3 4 5],
+                when: 'buy_train',
```

Comment:
```suggestion
                when: 'buying_train',
```

Reply:
I'm pretty sure we've established repeatedly that 'buying_train' leads to an issue where the associated ability doesn't work when it's the last remaining ability, or something of the sort.

Examples: 
https://github.com/tobymao/18xx/pull/10098
https://github.com/tobymao/18xx/pull/10318

Then again, other abilities in this game don't ever close, so this may be a non-issue for this game. What do you think?

Reply:
I was just going off the `when` values listed in ability/README.md. This has `buying_train` but not `buy_train`.

Looking at those issues you've linked this doesn't appear to have been fully resolved.

Reply:
well, annoyingly, this does seem to fix the issue entirely. So for the time being, I'm using this.